### ISSUE TITLE
Problem (?) with MFD in KinwaveImplicitOverlandFlow for certain DEMs

### ISSUE THREAD

**Issue by amanaster2** [landlab/landlab#2097]
Hello! I hope y'all have had a good holiday season :-)

Apologies in advance for the wall of text.

I've run into an interesting problem, and I was wondering if folks had any insight... @erkanistan and I were working on a teaching notebook to compare `KinwaveImplicitOverlandFlow` to a "brute force" calculation of runoff and discharge at the outlet of a specific watershed (North Fork Nooksack in WA). Ideally, these two methods should yield very similar results for maximum outlet discharge when using `flow_director = "MFD"`, and we confirmed this when using a different watershed (Spring Creek in CO). However, while using the NFN watershed, the "brute force" method yielded much higher (and much more reasonable) discharge at the outlet than `KinwaveImplicitOverlandFlow`. 

I thought that the issue had to do with sinks in the NFN DEM because of the MFD algorithm, so I pre-processed the DEM to fill the sinks (I tried this in QGIS, ArcGIS, and python), but no matter what pre-processing I tried, `KinwaveImplicitOverlandFlow` seemed to keep finding sinks!!

Below I provide the code used to compare the "brute force" method to `KinwaveImplicitOverlandFlow`.

```python
import numpy as np #import numpy
import landlab #import landlab

from landlab import RasterModelGrid #rastermodel sub classes
from landlab.io import read_esri_ascii #reading an esri ascii file 

from landlab.components import FlowAccumulator, KinwaveImplicitOverlandFlow #import our flow accumulator and kinematic wave component

%===============================
(rmg, Z) = read_esri_ascii('insert_dem_here.asc', name='topographic__elevation') #read in DEM
oid = rmg.set_watershed_boundary_condition('topographic__elevation', return_outlet_id=True) #finds the outlet and closes the watershed boundaries

%===============================
# BRUTE FORCE METHOD
# Instantiate Landlab FlowAccumulator using 'MFD' as the flow director, 2 mm/hr of runoff (converted to m/s), and
# use the partition method 'square_root_of_slope' to match instantiation of FlowAccumulator in Kinwave component
fa = FlowAccumulator(rmg,
                     surface='topographic__elevation',
                     flow_director='MFD', #multiple flow directions
                     runoff_rate=5.555556e-7, #2 mm/hr converted to m/s
                     partition_method='square_root_of_slope')

# Run method to get drainage area and discharge at each node
(drainage_area, discharge) = fa.accumulate_flow()

# Determine outlet drainage area
outlet_drainage_area = rmg.at_node['drainage_area'][oid]

# Obtain outlet discharge
outlet_discharge = rmg.at_node['surface_water__discharge'][oid]

# Calculate outlet at the outlet runoff in mm
runoff=(1000.*86400.*(rmg.at_node['surface_water__discharge'][oid]))/outlet_drainage_area

print('Outlet discharge (m^3/s) =', np.round(outlet_discharge,2))
print('Outlet runoff (mm) =', np.round(runoff,2))

%===============================
# KINWAVEIMPLICITOVERLANDFLOW METHOD
# Instantiate kinwave component
knwv = KinwaveImplicitOverlandFlow(rmg, runoff_rate=2, roughness=0.03, depth_exp=5/3) #Feed initial component a runoff rate of 2 mm/hr

# Initialize model run information
hydrograph_time = [0]
discharge_at_outlet = [0]
dt = 3600 #time step in seconds

elapsed_time = 1 #Set an initial time to avoid any 0 errors
model_run_time= 86400*1.5 #total model run time, in seconds; 36 hours
storm_duration = 86400 #length of storm in seconds; 24 hours

# Run the model; note that this will take a bit of time!
while elapsed_time <= model_run_time:
    if elapsed_time < storm_duration:
        knwv.run_one_step(dt)
    else:
        knwv.runoff_rate = 1e-30 #Reset runoff_rate to be ~0; post-storm runoff
        knwv.run_one_step(dt)

    q_at_outlet = rmg.at_node['surface_water_inflow__discharge'][oid].item() #get discharge at the outlet

    hydrograph_time.append(elapsed_time/3600.)
    discharge_at_outlet.append(q_at_outlet)
                        
    elapsed_time += dt #increase model time

print('Max outlet discharge = ' + str(np.max(np.round(discharge_at_outlet,2))) + ' m^3/s')
```

I'd be happy to provide the original .asc file for the NFN watershed, but I think I would need to do so outside of Github.

TLDR: When using MFD for both "brute force" and `KinwaveImplicitOverlandFlow` runoff calculations on a specific DEM (even pre-processed!!), `KinwaveImplicitOverlandFlow` yields unrealistic (i.e., much lower) discharge at the outlet. 

Have any of y'all ever run into this before? Is this just an issue with the DEM? Or is this a larger problem with `KinwaveImplicitOverlandFlow`?

**Comment by gregtucker** [comment]
@amanaster2 sorry for the slow reply! The notification got buried in my late-December email, and then the semester started...

Anyway, I haven't run into this before. Assuming I understand the comparison right, I agree that the steady state form of the kinwave solution should converge with a calculation using MFD (which assumes steady state from the get-go). Are you confident that your run with `KinwaveImplicit` hit steady state? You'd need to run it with storm duration >= basin concentration time.
